### PR TITLE
Add support for retrieving local bundles.

### PR DESCRIPTION
**Pull request by frankban**
With some limitations for now.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/36/
Test PASSed.

**Comment by fabricematrat** [comment]
:+1:  Code :ok:

File: local.go [hunk 1]
```diff
@@ -128,11 +121,49 @@ func (r *LocalRepository) Get(curl *charm.URL) (charm.Charm, error) {
...
+		return nil, entityNotFound(curl, r.Path)
+	}
+	return charm.ReadBundleDir(path)
+}
+
+// checkUrlAndPath checks that the given URL represents a local entity and that
+// the repository path exists.
+func (r *LocalRepository) checkUrlAndPath(curl *charm.URL) error {
+	if curl.Schema != "local" {
+		return fmt.Errorf("local repository got URL with non-local schema: %q", curl)
```
Comment (hatched) [inline local.go]: The charm schemas are confusing to many, especially new people. Could this return a message with the expected format.

Comment (hatched) [inline local.go]: I don't see  any tests for this failure case.

**Comment by hatched** [comment]
:+1: With a couple comments

File: local.go (hunk 1 above)
Comment (frankban) [inline local.go]: This is actually tested and the error is usually not user facing.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**